src/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java

```diff
@@ -704,63 +707,101 @@ public boolean onRightClick(ItemUseEvent e, Player p, ItemStack item) {
+													break;
+												}
+												else if (SlimefunManager.isItemSimiliar(inv.getContents()[j], converting, true, SlimefunManager.DataType.ALWAYS)) break;
+											}
 										}
-										else if (SlimefunManager.isItemSimiliar(inv.getContents()[j], converting, true, SlimefunManager.DataType.ALWAYS)) break;
 									}
-								}
-							}
 
-							if (craft) {
-								ItemStack adding = RecipeType.getRecipeOutputList(machine, inputs.get(i));
-								if (Slimefun.hasUnlocked(p, adding, true)) {
-									if (InvUtils.fits(inv, adding)) {
-										for (ItemStack removing: inputs.get(i)) {
-											if (removing != null) inv.removeItem(removing);
-										}
-										inv.addItem(adding);
-										p.getWorld().playSound(p.getLocation(), Sound.BLOCK_LAVA_POP, 1, 1);
-										p.getWorld().playEffect(b.getLocation(), Effect.MOBSPAWNER_FLAMES, 1);;
-										if (SlimefunStartup.chance(100, (Integer) Slimefun.getItemValue("SMELTERY", "chance.fireBreak"))) {
-											BlockBreaker.nullify(b.getRelative(BlockFace.DOWN).getRelative(BlockFace.DOWN));
+									if (craft) {
+										ItemStack adding = RecipeType.getRecipeOutputList(machine, inputs.get(i));
+										if (Slimefun.hasUnlocked(p, adding, true)) {
+											if (InvUtils.fits(inv, adding)) {
+												for (ItemStack removing: inputs.get(i)) {
+													if (removing != null) inv.removeItem(removing);
+												}
+												inv.addItem(adding);
+												p.getWorld().playSound(p.getLocation(), Sound.BLOCK_LAVA_POP, 1, 1);
+												p.getWorld().playEffect(b.getLocation(), Effect.MOBSPAWNER_FLAMES, 1);
+												Block raw_disp = b.getRelative(BlockFace.DOWN);									
+												Hopper resf = null;
+												
+												if(raw_disp.getRelative(BlockFace.EAST).getState().getBlock().getType().name() == "HOPPER") {
+													resf = (Hopper) raw_disp.getRelative(BlockFace.EAST).getState();
+												} else if (raw_disp.getRelative(BlockFace.WEST).getState().getBlock().getType().name() == "HOPPER") {
+													resf = (Hopper) raw_disp.getRelative(BlockFace.WEST).getState();
+												} else if (raw_disp.getRelative(BlockFace.NORTH).getState().getBlock().getType().name() == "HOPPER") {
+													resf = (Hopper) raw_disp.getRelative(BlockFace.NORTH).getState();
+												} else if (raw_disp.getRelative(BlockFace.SOUTH).getState().getBlock().getType().name() == "HOPPER"){
+													resf = (Hopper) raw_disp.getRelative(BlockFace.SOUTH).getState();
+												}
+												boolean isIgniteChamberInstalled = false;
+												
+												if(resf != null) {
+													if(resf.getCustomName().equalsIgnoreCase(SlimefunItems.IGNITION_CHAMBER.getItemMeta().getDisplayName())) isIgniteChamberInstalled = true;
+												}
+												
+												if (SlimefunStartup.chance(100, (Integer) Slimefun.getItemValue("SMELTERY", "chance.fireBreak"))) {
+													if(isIgniteChamberInstalled) {
```

Comment:
Please remove your boolean and use 

BlockStorage.check(block, "IGNITION_CHAMBER") 

instead.